### ISSUE TITLE
[3.3] Local extensions don't get config settings. 

### ISSUE THREAD

**Issue by bobdenotter** [bolt/bolt#6319]
Details
-------

 - Relevant Bolt Version: [ 3.3 | master ]

Reproduction
------------

- Set up this simple example: 

```
mkdir extensions/local
cd extensions/local
git clone git@github.com:bobdenotter/config-testing.git ConfigTesting
```

Add it to index.php

```
$app = require __DIR__ . '/app/web.php';
...
$app['extensions']->add(new \ConfigTesting\ConfigTestingExtension());
...
$app->run();
```

Go to 

You'll see this: 

![screen shot 2017-01-29 at 16 39 15](https://cloud.githubusercontent.com/assets/1833361/22405353/8acb8e42-e641-11e6-891f-8b4cb9f2f185.png)

Instead of a dump of "foo" and "bar".

See here: 

https://github.com/bobdenotter/config-testing/blob/master/ConfigTestingExtension.php#L43-L49

Additionally, I'd expect a new config file for the extension in `app/config/extensions/`.

**Comment by GawainLynch** [comment]
This is not a blocker.

Local extensions are for expert users and they are responsible for their own data/assets.

**Comment by rossriley** [comment]
OK, so this is fixed now, I've actually tested #6324 against this extension and it dumps the config as expected.

The only change that we made to the implementation is that we no longer magically autoload from extensions/local, it's up to you to define where your local extensions are installed. If you're happy with this location then just add this to your root composer.json:

```
        "psr-4" : {
            .............
            "": "extensions/local"
        }
```

**Comment by bobdenotter** [comment]
> If you're happy with this location then just add this to your root composer.json:

No probs.. I'll add it to the docs, and it'll be fine.
